components/workspacekit/cmd/rings.go

```diff
@@ -670,32 +672,32 @@ func copyResolvConf(ring2root string) error {
 	return nil
 }
 
-// generate <ring2root>/etc/hosts
-func generateHosts(ring2root string) error {
-	dst, err := os.OpenFile(filepath.Join(ring2root, "/etc/hosts"), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0644)
+func makeHostnameLocal(ring2root string) error {
+	hostname, err := os.Hostname()
 	if err != nil {
 		return err
 	}
-	defer dst.Close()
-
-	hostsStr := `127.0.0.1       localhost
-::1     localhost ip6-localhost ip6-loopback
-fe00::0 ip6-localnet
-fe00::0 ip6-mcastprefix
-fe00::1 ip6-allnodes
-fe00::2 ip6-allrouters
-`
-	hostname, err := os.Hostname()
-	if err == nil {
-		hostsStr += fmt.Sprintf("127.0.0.1 %s\n", hostname)
+	path := filepath.Join(ring2root, "/etc/hosts")
+	stat, err := os.Stat(path)
```

Comment:
if the file does not exist, we would not necessarily want to stop the workspace

Reply:
Yes, so this func only return error
https://github.com/gitpod-io/gitpod/blob/5b0af01cf950564d26e25aa525092cbfbe036300/components/workspacekit/cmd/rings.go#L353-L357

in here just log